src/pages/safe/apps.tsx

```diff
@@ -1,18 +1,33 @@
 import type { NextPage } from 'next'
 import Head from 'next/head'
+import { useRouter } from 'next/router'
 import Grid from '@mui/material/Grid'
 import { useSafeApps } from '@/hooks/safe-apps/useSafeApps'
 import { AppCard } from '@/components/safe-apps/AppCard'
 import { SafeAppsHeader } from '@/components/safe-apps/SafeAppsHeader'
+import AppFrame from '@/components/safe-apps/AppFrame'
+import SafeAppsErrorBoundary from '@/components/safe-apps/SafeAppsErrorBoundary'
+import SafeAppsLoadError from '@/components/safe-apps/SafeAppsErrorBoundary/SafeAppsLoadError'
 import { IS_PRODUCTION } from '@/config/constants'
 
 const Apps: NextPage = () => {
+  const router = useRouter()
+  const { appUrl } = router.query
+
   const { allSafeApps, remoteSafeAppsLoading, customSafeAppsLoading, addCustomApp } = useSafeApps()
 
   if (remoteSafeAppsLoading || customSafeAppsLoading) {
     return <p>Loading...</p>
   }
 
+  if (appUrl) {
+    return (
+      <SafeAppsErrorBoundary render={() => <SafeAppsLoadError />}>
+        <AppFrame appUrl={Array.isArray(appUrl) ? appUrl[0] : appUrl} />
```

Comment:
In what scenario will it be an array?

Reply:
`const { appUrl } = router.query` type is string | string[] and in a comment above Ivan suggested to avoid casting

Reply:
It's an array because in URLs you can do this: `google.com?search=cats&search=dogs`